src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -11561,12 +11561,14 @@ and TcLetBinding (cenv: cenv) isUse env containerInfo declKind tpenv (synBinds,
 
     // Canonicalize constraints prior to generalization
     let denv = env.DisplayEnv
-    CanonicalizePartialInferenceProblem cenv.css denv synBindsRange
-        (checkedBinds |> List.collect (fun tbinfo ->
-            let (CheckedBindingInfo(_, _, _, _, explicitTyparInfo, _, _, _, tauTy, _, _, _, _, _)) = tbinfo
-            let (ExplicitTyparInfo(_, declaredTypars, _)) = explicitTyparInfo
-            let maxInferredTypars = (freeInTypeLeftToRight g false tauTy)
-            declaredTypars @ maxInferredTypars))
+    try
+        CanonicalizePartialInferenceProblem cenv.css denv synBindsRange
+            (checkedBinds |> List.collect (fun tbinfo ->
+                let (CheckedBindingInfo(_, _, _, _, explicitTyparInfo, _, _, _, tauTy, _, _, _, _, _)) = tbinfo
+                let (ExplicitTyparInfo(_, declaredTypars, _)) = explicitTyparInfo
+                let maxInferredTypars = (freeInTypeLeftToRight g false tauTy)
+                declaredTypars @ maxInferredTypars))
+    with RecoverableException _ -> ()
```

Comment:
Please add more explanation as code comment why it is fine to gave an all-catching try block here, this might not be obvious for future readers.